lib/src/model_utils.dart

```diff
@@ -166,3 +169,150 @@ String crossdartifySource(
   }
   return newSource;
 }
+
+/// An UnmodifiableListView that computes equality and hashCode based on the
+/// equality and hashCode of its contained objects.
+class HashableList extends UnmodifiableListView<dynamic> {
+  HashableList(Iterable<dynamic> iterable) : super(iterable);
+
+  @override
+  bool operator ==(other) {
+    if (this.length == other.length) {
+      for (var index = 0; index < length; ++index) {
+        if (this[index] != other[index]) return false;
+      }
+      return true;
+    }
+    return false;
+  }
+
+  @override
+  get hashCode => hashObjects(this);
+}
+
+/// A type alias for [Tuple2]`<Function, HashableList>`.
+class _MemoKey extends Tuple2<Function, HashableList> {
```

Comment:
Do we really need this class?  It looks like every case of `new _MemoKey(f, new HashableList([param1, param2, ...])` could be replaced with `new HashableList([f, param1, param2, ...])`, and we would save a memory allocation.